satellite/satellitedb/overlaycache.go

```diff
@@ -303,76 +291,63 @@ func (cache *overlaycache) Update(ctx context.Context, info *pb.Node) (err error
 
 // Delete deletes node based on id
 func (cache *overlaycache) Delete(ctx context.Context, id storj.NodeID) error {
-	_, err := cache.db.Delete_OverlayCacheNode_By_NodeId(ctx,
-		dbx.OverlayCacheNode_NodeId(id.Bytes()),
-	)
+	_, err := cache.db.Delete_Node_By_Id(ctx, dbx.Node_Id(id.Bytes()))
 	return err
 }
 
-// Create a db entry for the provided storagenode
-func (cache *overlaycache) Create(ctx context.Context, nodeID storj.NodeID, startingStats *overlay.NodeStats) (stats *overlay.NodeStats, err error) {
+// CreateStats initializes the stats the provided storagenode
+func (cache *overlaycache) CreateStats(ctx context.Context, nodeID storj.NodeID, startingStats *overlay.NodeStats) (stats *overlay.NodeStats, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	var (
-		totalAuditCount    int64
-		auditSuccessCount  int64
-		auditSuccessRatio  float64
-		totalUptimeCount   int64
-		uptimeSuccessCount int64
-		uptimeRatio        float64
-		wallet             string
-		email              string
-	)
+	tx, err := cache.db.Open(ctx)
+	if err != nil {
+		return nil, Error.Wrap(err)
+	}
+	dbNode, err := tx.Get_Node_By_Id(ctx, dbx.Node_Id(nodeID.Bytes()))
+	if err != nil {
+		return nil, Error.Wrap(utils.CombineErrors(err, tx.Rollback()))
+	}
 
 	if startingStats != nil {
-		totalAuditCount = startingStats.AuditCount
-		auditSuccessCount = startingStats.AuditSuccessCount
-		auditSuccessRatio, err = checkRatioVars(auditSuccessCount, totalAuditCount)
+		auditSuccessRatio, err := checkRatioVars(startingStats.AuditSuccessCount, startingStats.AuditCount)
 		if err != nil {
 			return nil, errAuditSuccess.Wrap(err)
```

Comment:
If you started transaction each error return up to commit should call `Rollback()`